app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/hplus/HPlusSupport.java

```diff
@@ -636,6 +636,79 @@ public void onTestNewFunction() {
     }
 
 
+    private void showIncomingCall(String name, String number){
+        LOG.debug("Show Incoming Call");
+
+        try {
+            TransactionBuilder builder = performInitialized("incomingCallIcon");
+
+            //Enable call notifications
+            builder.write(ctrlCharacteristic, new byte[] {HPlusConstants.COMMAND_SET_INCOMING_CALL, 1 });
+
+            //Show Call Icon
+            builder.write(ctrlCharacteristic, HPlusConstants.COMMAND_ACTION_INCOMING_CALL);
+
+            //builder = performInitialized("incomingCallText");
+            builder.queue(getQueue());
+
+            try {
+                Thread.sleep(200);
```

Comment:
Have a look at the two `FetchActivityOperation` classes. For Mi1 you just ask to sync and the band tells you how much data you will get. For Mi2, you explicitly send a data from which you want to receive activity data. We look up the last sample in the database and use that timestamp. If there is none, we simply use current date - 10 days, IIRC.

Did you try sending the commands without the sleep()? I've seen other implementations using sleep() because they don't have proper synchronization of the BLE commands.